### PR TITLE
refactor: change the conditions to use router dealer

### PR DESCRIPTION
**Pull request by JoanFM**
Would require rebump of all hub executors

### CODE REVIEW

File: jina/peapods/peas/__init__.py [hunk 1]
```diff
@@ -404,11 +407,3 @@ def role(self) -> 'PeaRoleType':
 
         .. #noqa: DAR201"""
         return self.args.pea_role
-
-    @property
```
Comment (winstonww) [inline jina/peapods/peas/__init__.py:408]: Just for my understanding, what's the rationale for removing the `_is_inner_pea` property?

**Review by winstonww — APPROVED** [review]
LGTM

File: jina/peapods/peas/__init__.py (hunk 1 above)
Comment (JoanFM) [reply jina/peapods/peas/__init__.py:408]: that nobody is using and is private

Comment (winstonww) [reply jina/peapods/peas/__init__.py:408]: understood, thank you

**Pull request closed without merging**